packages/launchpad/src/default-wired-launchpad.tsx

```diff
@@ -10,12 +17,17 @@ type DefaultWiredLaunchpadProps = {
 	siteSlug: string | null;
 	checklistSlug: string;
 	launchpadContext: string;
+	extraActions?: {
+		siteLaunched?: () => void;
```

Comment:
At a structural level, `siteLaunched` isn't really an "action", and putting it in `extraActions` wasn't the best place to start with. Given that we're refactoring this part of the code/API, I think we should really aim to do two things:
1. Push the type for `extraActions` down into the base Launchpad types, and then explicitly exclude any types we don't want to allow as overrides in `DefaultWiredLaunchpad`, which itself may be best handled by an explicit type defined right next to the base type. That way we get new support actions by default in this component moving forward. We'll still need to exclude ones that we want excluded for this component, but that should be much easier to do with the relevant type a few lines down. (As-is, the details are in separate files that will both need separate maintenance.)
2. Move `siteLaunched` to its own prop, possibly called `onSiteLaunch` or similar.

Reply:
> Update the Share Site modal implementation to use the data stores as a first step
@daledupreez, about this point, do you mean the implementation that controls the visibility of the modal?

Reply:
Can you please check if this draft satisfies your concerns https://github.com/Automattic/wp-calypso/pull/82888

Reply:
> > Update the Share Site modal implementation to use the data stores as a first step
>
> @daledupreez, I got a little confused about this point. Do you mean the implementation that controls the visibility of the Share Site modal?

Now that I think about it, it feels like pushing that into a data store may be overkill. Your changes in #83095 move the state handling into the Launchpad package, which I think is the right starting point. Also, just to make sure that we have the right link, the state was being managed locally in the Customer Home card component rather than in the launchpad: https://github.com/Automattic/wp-calypso/blob/8bd338e7076e5351276588808c6062e0f0fc4121/client/my-sites/customer-home/cards/launchpad/index.tsx#L53

Reply:
> > At a structural level, siteLaunched isn't really an "action", and putting it in extraActions wasn't the best place to start with. Given that we're refactoring this part of the code/API, I think we should really aim to do two things:
>
> Can you please check if this draft satisfies your concerns https://github.com/Automattic/wp-calypso/pull/82888

Yep, that addresses them very neatly! I've just approved the PR.